onnxruntime/core/providers/nuphar/scripts/model_tools.py

```diff
@@ -0,0 +1,118 @@
+                if value_info.name == output:
+                    graph.value_info.remove(value_info)
+                    output_value_info = graph.output.add()
+                    output_value_info.CopyFrom(value_info)
+                    break
+
+    for node in model.graph.node:
+        if node.op_type == 'Loop':
+            # for debugging to make scan output as model graph output
+            set_op_output_as_model_output(node, model.graph)
+
+def run_with_ort(model_path, ort_test_case_dir=None):
+    def np_type_from_onnx_elem_type(elem_type):
+        if elem_type == 1:
+            return np.float32
+        else:
+            return None
+
+    def tensor_shape_from_onnx_shape(onnx_shape, dynamic_shape_cache, default_dynamic_dim):
+        tensor_shape = []
+        for dim in onnx_shape.dim:
+            if not dim.dim_param:
+                tensor_shape.append(dim.dim_value)
+            else:
+                if dynamic_shape_cache and dim.dim_param in dynamic_shape_cache:
+                    tensor_shape.append(dynamic_shape_cache[dim.dim_param])                    
+                else:
+                    tensor_shape.append(default_dynamic_dim)
+        return tensor_shape
+
+    def generate_tensor(input_type, dynamic_shape=None, default_dynamic_dim=30):
+        data_type = np_type_from_onnx_elem_type(input_type.tensor_type.elem_type)
+        tensor_shape = tensor_shape_from_onnx_shape(input_type.tensor_type.shape, dynamic_shape, default_dynamic_dim)
+        return np.random.rand(*tensor_shape).astype(data_type)
+
+    model = onnx.load(model_path)
+    input_names = [input.name for input in model.graph.input]
+            
+    np.random.seed(0)
+    inputs = [generate_tensor(input.type) for input in model.graph.input]
+    input_dict = dict(zip(input_names, inputs))
+
+    session = ort.InferenceSession(model.SerializeToString())
+    outputs = session.run(None, input_dict) # {'input': input, 'hi0': hi0, 'ci0': ci0}
+
+    if ort_test_case_dir:
+        def save_ort_test_case(ort_test_case_dir):
+            for i, (input_name, input) in enumerate(input_dict.items()):
+                SaveTensorProto(os.path.join(test_data_dir, 'input_{0}.pb'.format(i)),
+                                input_name, input)
+
+            output_names = [output.name for output in model.graph.output]
+            output_dict = dict(zip(output_names, outputs))
+            for i, (output_name, output) in enumerate(output_dict.items()):
+                SaveTensorProto(os.path.join(test_data_dir, 'output_{0}.pb'.format(i)), output_name, output)
+
+        save_ort_test_case(ort_test_case_dir)
+
+    return outputs
+
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--tool', help='what to do',
+                        choices=['save_model_sub_graph',
+                                 'run_shape_inference',
+                                 'run_with_ort'])
```

Comment:
>run_with_ort [](start = 34, length = 12)

Use this? [https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/tools/onnxruntime_test.py](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/tools/onnxruntime_test.py)